common/worker.c

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2017 Drew Walters <drewwalters96@gmail.com>
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <error.h>
+#include <worker.h>
+
+void
+queue_worker(worker worker_func)
+{
+	/* Add worker function to available queue index */
+	for (size_t i = 0; i < MAX_WORKERS; i++) {
+		if (workers[i] != NULL) {
+			continue;
+		} else {
+			workers[i] = worker_func;
+			return;
+		}
+
+		panic("Worker function queue is full")
+	}
+}
+
+void
+process_workers(void)
+{
+	for (size_t i = 0; i < MAX_WORKERS; i++) {
+		if (workers[i] == NULL) {
+			continue;
+		} else {
+			/* Call worker and check result */
+			if (!(*workers[i])(ptr))
```

Comment:
`if (!workers[i](ptr))` is sufficient, but as we discussed, it's simpler to have the worker itself panic, so this simplifies even further to `workers[i](ptr);`. (um, `ptr` isn't declared anywhere yet...).

Reply:
> if (!workers[i](ptr)) is sufficient, but as we discussed, it's simpler to have the worker itself panic, so this simplifies even further to workers[i](ptr);

Done. The `typedef` has also changed.

> (um, ptr isn't declared anywhere yet...).
WIP :relaxed:. This only acting as a placeholder for a change that will be pushed soon.